frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/switch-to-form-validation-utils.ts

```diff
@@ -0,0 +1,110 @@
+import i18n from 'i18next';
+import * as yup from 'yup';
+import { nameValidationSchema } from '@console/shared';
+import { PipelineTask } from '../../../types';
+import { PipelineBuilderFormValues } from './types';
+import { runAfterMatches } from './validation-utils';
+
+const resourceDefinitionYAML = () => {
+  return yup.array().of(
+    yup.object({
+      name: yup.string(),
+      resource: yup.string(),
+    }),
+  );
+};
+
+export const validRunAfter = (formData: PipelineBuilderFormValues, thisTask: PipelineTask) => {
+  return yup.array().of(
+    yup
+      .string()
+      .test('tasks-matches-runAfters', i18n.t('pipelines-plugin~Invalid runAfter'), function(
+        runAfter: string,
+      ) {
+        return runAfterMatches(formData, [runAfter], thisTask.name);
+      }),
+  );
+};
+
+const taskValidationYAMLSchema = (formData: PipelineBuilderFormValues) => {
+  return yup.array().of(
+    yup.lazy((taskObject) =>
+      yup
+        .object({
+          name: nameValidationSchema((tKey) => i18n.t(tKey)),
+          taskRef: yup
+            .object({
+              name: yup.string(),
+              kind: yup.string(),
+            })
+            .default(undefined),
+          taskSpec: yup.object(),
+          runAfter: validRunAfter(formData, taskObject),
+          params: yup.array().of(
+            yup.object({
+              name: yup.string(),
+              value: yup.lazy((value) => {
+                if (Array.isArray(value)) {
+                  return yup.array().of(yup.string());
+                }
+                return yup.string();
+              }),
+            }),
+          ),
+          resources: yup.object({
+            inputs: resourceDefinitionYAML(),
+            outputs: resourceDefinitionYAML(),
+          }),
+          when: yup.array().of(
+            yup.object({
+              input: yup.string(),
+              operator: yup.string(),
+              values: yup.array().of(yup.string()),
+            }),
+          ),
+
+          workspaces: yup.array().of(
+            yup.object({
+              name: yup.string(),
```

Comment:
To solve the above issue, I think you should make this one required field, so the safeOmit removes the invalid path.

```suggestion
              name: yup.string().required(),
```

Reply:
done.